stable_baselines3/common/off_policy_algorithm.py

```diff
@@ -487,20 +493,29 @@ def _store_transition(
 
         # As the VecEnv resets automatically, new_obs is already the
         # first observation of the next episode
-        if done and infos[0].get("terminal_observation") is not None:
-            next_obs = infos[0]["terminal_observation"]
-            # VecNormalize normalizes the terminal observation
-            if self._vec_normalize_env is not None:
-                next_obs = self._vec_normalize_env.unnormalize_obs(next_obs)
-        else:
-            next_obs = new_obs_
+        next_obs = deepcopy(new_obs_)  # use deepcopy for dict observations
```

Comment:
`deepcopy` is only necessary for dict obs, yes? Might be worth saving the processing time by moving it inside the dict-obs check or so.

Reply:
Yup. This sounds a lot like premature optimization, but to my experience it is the copies that bottleneck performance, so might be worth saving up here :)